packit/schema.py

```diff
@@ -467,22 +476,8 @@ class PackageConfigSchema(CommonConfigSchema):
     @pre_load
     def ordered_preprocess(self, data, **kwargs):
         data = self.rename_deprecated_keys(data, **kwargs)
-        data = self.add_defaults_for_jobs(data, **kwargs)
-        return data
-
-    @staticmethod
-    def add_defaults_for_jobs(data, **_):
-        """
-        add all the fields (except for jobs) to every job so we can process only jobconfig in p-s
-        """
-        if not data:  # data is None when .packit.yaml is empty
-            return data
-        for job in data.get("jobs", []):
-            for k, v in data.items():
-                if k == "jobs":
-                    # overriding jobs doesn't make any sense
-                    continue
-                job.setdefault(k, v)
+        data.setdefault("jobs", get_default_jobs())
```

Comment:
ahaaaa, so it moved here

Reply:
Yes. This might be a little bit purist, but what I've tried to achieve here, is to move all code that deals with the schema and defaults to `PackageConfigSchema`. Only things that require some other information than the content of the config are set in `get_from_dict()`.

Reply:
Thanks for the explanation!

I can see now that you mentioned that in the respective commit message: 6d0365eaa6614a223c60a30ba369178eebf00062

I wonder if it makes sense to capture this decision in the schema.py module docstring to guide future contributions.

Reply:
I put this in the doc-string of `PackageConfigSchema`. Please have a look and let me know if it needs tweaking.